conans/client/hook_manager.py

```diff
@@ -58,11 +62,19 @@ def execute(self, method_name, **kwargs):
         assert method_name in valid_hook_methods, \
             "Method '{}' not in valid hooks methods".format(method_name)
         for name, method in self.hooks[method_name]:
+            # Protect from recursive interprocess calls
+            lockfile = os.path.join(self._hooks_folder, name + ".lock")
+            lock = fasteners.InterProcessLock(lockfile, logger=logger)
+            gotten = lock.acquire(blocking=False)
+            if not gotten:
```

Comment:
This is really dangerous, can skip hooks execution for independent Conan processes, doing different things that otherwise would not conflict at all. The solution should be probably based on logic, no interprocess locking.

It is also possible that InterProcessLock doesn't behave as expected for child processes, for example it doesn't work with threads at all.

Reply:
Indeed, I mentioned it in the commentary here https://github.com/conan-io/conan/pull/6434#issuecomment-599637014

The problem with logic is that I don't see how we can protect subprocesses from, for example, `os.system()`.  But what we can do about it?

Reply:
Oh, I only just now noticed that there is an abstraction of the InterProcessLock - ReadLock/WriteLock in locks.